### PR TITLE
Update rest error keys to match Android

### PR DESCRIPTION
**Pull request by bbirman**
Related to https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2095

### CODE REVIEW

File: ios/SalesforceReact/SFNetReactBridge.m [hunk 1]
```diff
@@ -105,10 +105,11 @@ @implementation SFNetReactBridge
...
                                       failureBlock:^(id response, NSError *e, NSURLResponse *rawResponse) {
                                           __strong typeof(self) strongSelf = weakSelf;
+                                          NSMutableDictionary *responseDictionary = [[rawResponse asDictionary] mutableCopy];
+                                          responseDictionary[@"body"] = [strongSelf serializableResponse:response rawResponse:rawResponse];
                                           NSMutableDictionary *errorDictionary = [NSMutableDictionary new];
                                           errorDictionary[@"error"] = e.localizedDescription;
-                                          errorDictionary[@"urlResponse"] = [rawResponse asDictionary];
-                                          errorDictionary[@"response"] = [strongSelf serializableResponse:response rawResponse:rawResponse];
+                                          errorDictionary[@"response"] = responseDictionary;
                                           callback(@[RCTMakeError(@"sendRequest failed", nil, errorDictionary)]);
```
Comment (bbirman) [inline ios/SalesforceReact/SFNetReactBridge.m:113]: Android returns a string currently while RCTMakeError gives a dictionary, should we change one of them?

Comment (bhariharan) [reply ios/SalesforceReact/SFNetReactBridge.m:113]: Let's use the same approach as what we did for hybrid?

Comment (bhariharan) [reply ios/SalesforceReact/SFNetReactBridge.m:113]: Maybe we should leave this as-is for now. We're very close to releasing `8.2`. Maybe we can line them up in `8.3`?

Comment (wmathurin) [reply ios/SalesforceReact/SFNetReactBridge.m:113]: On Android, we always send back strings (for success and error callbacks). The js side takes care of parsing the string into an object. See https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative/blob/dev/src/react.force.common.js#L63

Comment (bbirman) [reply ios/SalesforceReact/SFNetReactBridge.m:113]: Ahh, thanks!

**Pull request merged**